README.md

```diff
@@ -1,44 +1,235 @@
-# COMP110 Worksheet 4
+COMP110 Worksheet 4
+===================
 
 Please edit this README.md file with your answers to the worksheet questions.
 
-## Question 1
-
-### a
-
-### b
-
-### c
-
-### d
-
-## Question 2
-
-### a
-
-### b
-
-### c
-
-### d
-
-## Question 3
-
-### a
-
-### b
-
-### c
-
-### d
-
-## Question 4
-
-### a
-
-### b
-
-### c
-
-### d
-
+Question 1 and Question 2
+-------------------------
+
+### A
+
+| A | B | C | Output |
+|---|---|---|--------|
+| 0 | 0 | 0 | 0      |
+| 1 | 0 | 0 | 0      |
+| 0 | 1 | 0 | 0      |
+| 1 | 1 | 0 | 1      |
+| 0 | 0 | 1 | 0      |
+| 1 | 0 | 1 | 0      |
+| 0 | 1 | 1 | 0      |
+| 1 | 1 | 1 | 0      |
+
+![A picture containing wall, indoor Description automatically generated](media/ac9a2ad20ed3f3b001a506e47e760624.jpg)
+
+A picture containing wall, indoor Description automatically generated
+
+A picture containing wall, indoor Description automatically generated
+
+A picture containing wall, indoor Description automatically generated
+
+### B
+
+| A | B | C | Output |
+|---|---|---|--------|
+| 0 | 0 | 0 | 0      |
+| 1 | 0 | 0 | 1      |
+| 0 | 1 | 0 | 0      |
+| 1 | 1 | 0 | 1      |
+| 0 | 0 | 1 | 0      |
+| 1 | 0 | 1 | 1      |
+| 0 | 1 | 1 | 0      |
+| 1 | 1 | 1 | 0      |
+
+![](media/83440000c2b601f6f6e153a82e50c507.png)
+
+### C
+
+| A | B | C | Output |
+|---|---|---|--------|
+| 0 | 0 | 0 | 0      |
+| 1 | 0 | 0 | 1      |
+| 0 | 1 | 0 | 0      |
+| 1 | 1 | 0 | 1      |
+| 0 | 0 | 1 | 1      |
+| 1 | 0 | 1 | 1      |
+| 0 | 1 | 1 | 0      |
+| 1 | 1 | 1 | 1      |
+
+![A close up of a device Description automatically generated](media/d705e1a7c2583d448478092c3c4c695e.jpg)
+
+A close up of a device Description automatically generated
+
+A close up of a device Description automatically generated
+
+\~ A close up of a device Description automatically generated
+
+### D
+
+| A | B | C | D | Output |
+|---|---|---|---|--------|
+| 0 | 0 | 0 | 0 | 0      |
+| 1 | 0 | 0 | 0 | 0      |
+| 0 | 1 | 0 | 0 | 0      |
+| 1 | 1 | 0 | 0 | 0      |
+| 0 | 0 | 1 | 0 | 0      |
+| 1 | 0 | 1 | 0 | 0      |
+| 0 | 1 | 1 | 0 | 0      |
+| 1 | 1 | 1 | 0 | 0      |
+| 0 | 0 | 0 | 1 | 0      |
+| 1 | 0 | 0 | 1 | 0      |
+| 0 | 1 | 0 | 1 | 0      |
+| 1 | 1 | 0 | 1 | 0      |
+| 0 | 0 | 1 | 1 | 0      |
+| 1 | 0 | 1 | 1 | 0      |
+| 0 | 1 | 1 | 1 | 0      |
+| 1 | 1 | 1 | 1 | 0      |
+
+![](media/2e68faabff5599934eb4d7fbab4268dd.png)
+
+Question 3
+----------
+
+### A
+
+| A  | B  | Output   |
+|----|----|----------|
+| 0  | 0  | 0        |
+| 1  | 0  | 1        |
+| 0  | 1  | 1        |
+| 1  | 1  | 1        |
```

Comment:
You're missing a `not` on these values -- then they would match the second set of values